corehq/apps/linked_domain/updates.py

```diff
@@ -540,6 +544,102 @@ def update_hmac_callout_settings(domain_link):
     model.save()
 
 
+def update_auto_update_rule(domain_link, id):
+    upstream_rule_definition = local_get_auto_update_rule(domain_link.master_domain, id)
+    downstream_rule = get_or_create_downstream_rule(domain_link.linked_domain, upstream_rule_definition)
+    _update_rule(downstream_rule, upstream_rule_definition)
+
+
+def get_or_create_downstream_rule(domain, upstream_definition):
```

Comment:
Could `AutomaticUpdateRule.objects.get_or_create` be used instead of this function?

Reply:
How important is avoiding boilerplate vs avoiding database calls? It seems `get_or_create` results in an unnecessary `save` call, and I'd have to shift around transaction logic to account for that.